components/BLE/COMPONENT_BlueNRG_2/BlueNrg2HCIDriver.cpp

```diff
@@ -0,0 +1,632 @@
+/*
+ * Copyright (c) 2017-2020 ARM Limited
+ * Copyright (c) 2017-2020 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+
+// drivers
+#include "drivers/DigitalOut.h"
+#include "drivers/SPI.h"
+#include "drivers/InterruptIn.h"
+
+// platform
+#include "platform/mbed_wait_api.h"
+
+// FEATURE_BLE/targets/TARGET_CORDIO
+#include "CordioBLE.h"
+#include "CordioHCIDriver.h"
+#include "CordioHCITransportDriver.h"
+#include "hci_api.h"
+#include "hci_cmd.h"
+#include "hci_core.h"
+#include "dm_api.h"
+#include "bstream.h"
+#include "hci_mbed_os_adaptation.h"
+
+// rtos
+#include "rtos/Thread.h"
+#include "rtos/Semaphore.h"
+#include "rtos/Mutex.h"
+
+#define HCI_RESET_RAND_CNT              4
+
+#define VENDOR_SPECIFIC_EVENT           0xFF
+#define EVT_BLUENRG_2_INITIALIZED       0x0001
+#define ACI_READ_CONFIG_DATA_OPCODE     0xFC0D
+#define ACI_WRITE_CONFIG_DATA_OPCODE    0xFC0C
+#define ACI_GATT_INIT_OPCODE            0xFD01
+#define ACI_GAP_INIT_OPCODE             0xFC8A
+
+#define PUBLIC_ADDRESS_OFFSET           0x00
+#define RANDOM_STATIC_ADDRESS_OFFSET    0x80
+#define LL_WITHOUT_HOST_OFFSET          0x2C
+
+#define SPI_STACK_SIZE                  1024
+
+#define IRQ_TIMEOUT_DURATION            15 //ms
+
+namespace ble {
+namespace vendor {
+namespace bluenrg_2 {
+
+/**
+ * BlueNRG_2 HCI driver implementation.
+ * @see cordio::CordioHCIDriver
+ */
+class HCIDriver : public cordio::CordioHCIDriver {
```

Comment:
After the BLE refactor (https://github.com/ARMmbed/mbed-os/pull/13365) is merged this will need to drop the cordio:: namespace as it doesn't exists any more, now that cordio is *the* BLE stack
```suggestion
class HCIDriver : public CordioHCIDriver {
```

Reply:
Basically all files will need two search and replace done that remove completely these two strings:
vendor::cordio::
cordio::